frontend/src/pages/AllChatsPage.jsx

```diff
@@ -23,16 +23,50 @@ const ChatInboxHeader = () => (
     </div>
 )
 
+// find the blocked accounts of the user
+function useFetchBlockedUsers(myUid) {
+    const [res] = useBackendFetchJson({
+        route: `/chat/block/user/${myUid}`,
+        deps: [myUid],
+        isValid: myUid,
+    })
+    const blockedUsers = res?.blocks || []
+    // create a blocked user dictionary
+    const reduceList = blockedUsers.reduce((accumulator, item) => {
+        if (item.blocked === true) {
+            accumulator[item.uid] = item.blocked
+        }
+        return accumulator
+    }, {})
+    return reduceList
+}
+
 export default function AllChatsPage() {
     const authUser = useCurrentAuthUser()
     const matches = useGetAllUserMatches(authUser?.uid)
+    const blockedUsers = useFetchBlockedUsers(authUser?.uid)
+
+    const unblockedMatches = matches.filter(
+        (match) =>
+            !Object.keys(match.participants).some((participant) =>
+                Object.keys(blockedUsers).includes(participant)
+            )
+    )
+
+    const blockedMatches = matches.filter((match) =>
+        Object.keys(match.participants).some((participant) =>
+            Object.keys(blockedUsers).includes(participant)
+        )
+    )
```

Comment:
Optional: Could we make this code simpler? Right now, scanning the code it looks like:

```js
const unblockedMatches = blah blah blah
  blah blah blah
  blah blah blah
  blah blah blah

const blockedMatches = blah blah blah
  blah blah blah
  blah blah blah
  blah blah blah
```

Which is not bad, at least we can tell what is going on based on the variables names. But the logic for each variable has a lot going on.

Here are three ideas, which do you like better?

Option A: Add a `hasBlockedUser` field to each match object, then filter on it

```js
const matchesWithBlocks = matches.map(...)
const unblockedMatches = matches.filter(m =>! m.hasBlockedUser)
const blockedMatches = matches.filter(m => m.hasBlockedUser)
```

Option B: Create a helper function `hasBlockedUser(match)` to handle filtering

```js
const unblockedMatches = matches.filter(m =>!hasBlockedUser(m))
const blockedMatches = matches.filter(m => hasBlockedUser(m))
```

Option C: Keep the code as-is, with each array having its own, separate filtering logic

Reply:
To me, **Option B** looks best as we won't need to add additional fields into the `matches` object and we can filter out any matches with block users based on the helper function.